### DISCUSSION TITLE
Allow no window decoration together with apps (as exists in iTerm)

### DISCUSSION THREAD

**Discussion by cristeahub** [ghostty-org/ghostty#3148] [Feature Requests, Ideas] [answered]
Hello! Thanks for creating ghostty.

I am migrating from iTerm and there I am currently using a no window decoration + tabs mode style. See images:

Default:
<img width="962" alt="image" src="https://github.com/user-attachments/assets/9823b7c2-d05b-4c32-8cf6-1f3f8608aade" />
After pressing cmd+t:
<img width="962" alt="image" src="https://github.com/user-attachments/assets/3bbf756b-19ac-41a1-b149-1b9b8d67f904" />

Currenlty this configuration is not possible in Ghostty, as `window-decoration = false` does not allow for tabs to be created.

I note that on iTerm the tab bar is added to the window height after adding a new tab. This would suggest to me that the window decoration is created when pressing cmd+t. The most important thing would be to have small tabs to keep it visually pleasing.

To allow for this feature to be created one would make the current config make sense:

```
window-decoration = false
macos-titlebar-style = tabs
```

**Comment by mitchellh** [comment, marked as answer]
This has been previously discussed. This will require a custom tab bar implementation (iTerm2 uses a custom tab bar). We use native macOS tabs which require a titlebar. I'd like to build a custom tab bar solution but its not on the immediate roadmap.
